code/modules/antagonists/heretic/heretic_knowledge.dm

```diff
@@ -158,7 +177,7 @@
 			var/obj/item/stack/sac_stack = sacrificed
 			var/how_much_to_use = 0
 			for(var/requirement in required_atoms)
-				if(istype(sacrificed, requirement))
+				if(istype(sacrificed, requirement) || (islist(requirement) ? is_type_in_list(sacrificed, requirement) : FALSE))
 					how_much_to_use = min(required_atoms[requirement], sac_stack.amount)
 					break
```

Comment:
I don't really like the ternary `x ? y : FALSE` shenanigans for readability, if we could separate this out:
```dm
if(islist(requirement) && !is_type_in_list(sacrificed, requirement))
    continue
if(!istype(sacrificed, requirement)
    continue
how_much_to_use = min(required_atoms[requirement], sac_stack.amount)
break
```